pyatlan/generator/templates/methods/asset/connection.jinja2

```diff
@@ -9,30 +9,30 @@
         admin_users: Optional[list[str]] = None,
         admin_groups: Optional[list[str]] = None,
         admin_roles: Optional[list[str]] = None,
+        client: Optional[AtlanClient] = None,
```

Comment:
If we are going to add a client as a parameter then I think it should be in Referenceable, but I don't really like the need this dependence on a client.

Reply:
I don't either — for me it comes down to when we do the validation:

- At the time of object creation (advantage is we fail fast)
- At the time of object marshalling (just before sending to Atlan) — would probably be the more logical point to inject the client, but means any potential failure is delayed